src/internal/kopia/base_finder.go

```diff
@@ -29,22 +29,79 @@ const (
 	userTagPrefix = "tag:"
 )
 
-type Reason struct {
-	ResourceOwner string
-	Service       path.ServiceType
-	Category      path.CategoryType
+// TODO(ashmrtn): Move this into some inject package. Here to avoid import
+// cycles.
+type Reason interface {
+	Tenant() string
+	Resource() string
+	Service() path.ServiceType
+	Category() path.CategoryType
+	SubtreePath() (path.Path, error)
```

Comment:
This is a little opaque for those of us who aren't as familar with kopia.  Can you please give this a comment explaining exactly what should be expected from the value?